src/commands/Admin/Verifier.ts

```diff
@@ -407,4 +346,56 @@ export default class Verifier extends Command {
         await buttonInteraction.followUp({ content: `You are registered to the current semester, ${currentSemesterName}.`, flags: MessageFlags.Ephemeral });
         
     }
+
+
+    public async handleButtonInteractions(buttonInteraction: ButtonInteraction): Promise<void> {
+        const verifiedRole: Role = (await buttonInteraction.guild?.roles.fetch(this.verifiedRoleId))!;
+
+        if (verifiedRole.members.has(buttonInteraction.user.id)) {
+            buttonInteraction.reply({ content: "Your account is already linked to an email in our system. To link your discord account to another email address, use the `/logout` command first.", flags: MessageFlags.Ephemeral });
+            return;
+        }
+
+        // verification limit check
+        let remainingAttempts: number = (await VerificationAttempts.findAll({
+            where: {
+                discord_client_id: buttonInteraction.user.id,
+            },
+        }))[0]?.dataValues.remaining_failed_attempts;
+        remainingAttempts = remainingAttempts == undefined ? defaultFailedAttemptLimit : remainingAttempts;
+
```

Comment:
This entire function should be moved out of this command and into its own module, like `handlers` or something similar